### PR TITLE
feat: added workflow step to deploy the canary image to prod

### PR DESCRIPTION
**Pull request by kumarnarendra701**
**- Added workflow step to deploy the canary image to prod**

### CODE REVIEW

File: at_secondary/Dockerfile.canary_to_prod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM atsigncompany/secondary:canary
+ENV HOMEDIR=/atsign
+RUN \
+  cp -rpf pubspec.yaml $HOMEDIR/; \
+  chown -R atsign:atsign $HOMEDIR ;
```
Comment (cpswan) [inline at_secondary/Dockerfile.canary_to_prod:5]: I know that we have ENTRYPOINT defined in the underlying image, but do we need it defined here too?

**Review by cpswan — APPROVED** [review]
Generally LGTM. Just one question on ENTRYPOINT before we merge.

Any thoughts on how we test this?

**Comment by cpswan** [comment]
@gkc and @kalluriramkumar I've added you as reviewers as I'd like your eyes on this before we merge.

**Review by sitaram-kalluri — APPROVED** [review]

File: .github/workflows/at_server.yaml [hunk 2]
```diff
@@ -711,6 +726,125 @@ jobs:
...
+  # Rollback prod secondary image.
+  rollback_prod_secondary_image:
+    if: ${{ github.event.inputs.rollback_prod_secondary_image == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.0.2
+      
+      # Extract version for docker tag
+      - name: Get version
+        run: echo "VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
```
Comment (gkc) [inline .github/workflows/at_server.yaml:739]: Am I correct that when rolling back, we will run this job from the branch we want to rollback **to**?

File: .github/workflows/at_server.yaml [hunk 3]
```diff
@@ -711,6 +726,125 @@ jobs:
...
+    env:
+      working-directory: at_server
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3.0.2
+
+      # Extract version for docker tag
+      - name: Get version
+        run: echo "VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
```
Comment (gkc) [inline .github/workflows/at_server.yaml:820]: Which branch would we run this workflow on?

**Review by gkc — COMMENTED** [review]
I have asked a couple of questions to clarify my understanding of how this all works, but LGTM

File: .github/workflows/at_server.yaml (hunk 2 above)
Comment (cpswan) [reply .github/workflows/at_server.yaml:739]: It has nothing to do with branches here, as it's retagging Docker images.

File: .github/workflows/at_server.yaml (hunk 3 above)
Comment (cpswan) [reply .github/workflows/at_server.yaml:820]: When running it we pick a tag for a previous release.

**Pull request merged**